slice bounds out of range on AdminAccess role

Given the below IAM Policy JSON:
```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": "*",
            "Resource": "*"
        }
    ]
}
```
`awsiam` throws when filter is applied (ex `awsiam aws-controltower-AdministratorExecutionRole s3`):
```shell
panic: runtime error: slice bounds out of range [:2] with length 1

goroutine 1 [running]:
main.FilterPolicies(...)
        /home/giermulnik/work/src/CODE/awsiam/awsiam/main.go:234
main.main()
        /home/giermulnik/work/src/CODE/awsiam/awsiam/main.go:275 +0x874
```

--- a/awsiam/main.go
+++ b/awsiam/main.go
@@ -212,7 +212,14 @@ func GetPoliciesForRole(ctx context.Context, svc *iam.Client, roleName *string) 
 		}
 
 		var doc Document
-		if err := json.Unmarshal([]byte(*p.PolicyDocument), &doc); err != nil {
+		var decodedPolicy string
+
+		decodedPolicy, err = url.QueryUnescape(*p.PolicyDocument)
+		if err != nil {
+			return nil, fmt.Errorf("failed to URL-decode role policy document, %w", err)
+		}
+
+		if err := json.Unmarshal([]byte(decodedPolicy), &doc); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal role policy document, %w", err)
 		}
 
@@ -231,7 +238,7 @@ func FilterPolicies(policies PolicyDocuments, filter string) PolicyDocuments {
 		for _, stmt := range doc.Statement {
 			actions := make([]string, 0)
 			for _, action := range stmt.Action {
-				if action == filter || action[:len(filter)] == filter {
+				if action == filter || (len(action) > 1 && action[:len(filter)] == filter) {
 					actions = append(actions, action)
 				}
 			}

